inotify/adapters.py

```diff
@@ -227,53 +227,55 @@ def event_gen(self):
 
 class InotifyTree(BaseTree):
     def __init__(self, path, mask=inotify.constants.IN_ALL_EVENTS,
-                 block_duration_s=_DEFAULT_EPOLL_BLOCK_DURATION_S):
+                 block_duration_s=_DEFAULT_EPOLL_BLOCK_DURATION_S, ignored_dirs=[]):
         super(InotifyTree, self).__init__(mask=mask, block_duration_s=block_duration_s)
 
         self.__root_path = path
 
-        self.__load_tree(path)
+        self.__load_tree(path, ignored_dirs)
 
-    def __load_tree(self, path):
+    def __load_tree(self, path, ignored_dirs):
         _LOGGER.debug("Adding initial watches on tree: [%s]", path)
 
         q = [path]
         while q:
             current_path = q[0]
             del q[0]
+            
+            if current_path not in ignored_dirs:
+                self._i.add_watch(current_path, self._mask)
 
-            self._i.add_watch(current_path, self._mask)
-
-            for filename in os.listdir(current_path):
-                entry_filepath = os.path.join(current_path, filename)
-                if os.path.isdir(entry_filepath) is False:
-                    continue
+                for filename in os.listdir(current_path):
+                    entry_filepath = os.path.join(current_path, filename)
+                    if os.path.isdir(entry_filepath) is False:
+                        continue
 
-                q.append(entry_filepath)
+                    q.append(entry_filepath)
 
 
 class InotifyTrees(BaseTree):
 
     def __init__(self, paths, mask=inotify.constants.IN_ALL_EVENTS,
-                 block_duration_s=_DEFAULT_EPOLL_BLOCK_DURATION_S):
+                 block_duration_s=_DEFAULT_EPOLL_BLOCK_DURATION_S, ignored_dirs=[]):
         super(InotifyTrees, self).__init__(mask=mask, block_duration_s=block_duration_s)
```

Comment:
You should push ignored_dirs down to (_)BaseTree's __init__ and store it on object from there as it needs to be referenced there (line 216).